From 7c7cdc21232f0b3381ed8b78380dc7fb9a8b7269 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Sun, 01 Mar 2020 08:05:02 +0200
Subject: [PATCH] [SW-8229] goya2: remove security bypass strap

Bypassing security isn't dependent on bootstrap, only on efuse.
The bootstrap is used by hardware to control running legacy of
secureboot.

Change-Id: I222d89b7f7f6428573456564f4bf179642d352a9
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/common/secureboot.c b/common/secureboot.c
index 622baf4..63482aa 100644
--- a/common/secureboot.c
+++ b/common/secureboot.c
@@ -52,12 +52,7 @@
 	/* check efuse for security validation enable */
 	config_fuse = sp_sram_readl(SECURITY_EFUSE_SYS_CFG_OFFSET);
 	if ((config_fuse & EFUSE_SYSCFG_ENABLE_VALIDATION_MASK) == 0)
-		/* if efuse isn't set, check status of security bypass
-		 * boot strap if enabled bypass security validation.
-		 */
-		if ((regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS_H) &
-		     PSOC_GLOBAL_CONF_BOOT_STRAP_PINS_H_SECURITY_BYPASS_MASK))
-			return SECUREBOOT_VERIFY_SUCCESS;
+		return SECUREBOOT_VERIFY_SUCCESS;
 
 	/* check minimal valid security FW version */
 	if (security_check_valid_fw_version(fw_version) !=
